### DISCUSSION TITLE
Explain better what non generational updates mean

### DISCUSSION THREAD

**Discussion by m3thm4th** [gmodena/nix-flatpak#100] [Q&A] [answered]
In the [Readme](https://github.com/gmodena/nix-flatpak/blob/8db1648a95446d41e97ace7caf54e172669810ff/README.md?plain=1#L168) it talks about the fact that updates are not generational, does this mean that every time `nixos-rebuild switch` is run all the gigantic flatpaks are downloaded and written to disk once again, even when there's no update for them?

**Comment by gmodena** [comment, marked as answer]
flatpak applications are not re-installed at each activation, unless updates are available. It's however possible to pin a specific version (and avoid updating) via the  `commit=<hash>` attribute.

By "non-generational," it means that `nix-flatpak` does not inherently keep track of, or manage, multiple versions (generations) of a package over time like Nix does. This means that if you `rollback` your system (or home manager), you may not get the same versions of flatpak applications you had before the rollback.

**Reply by m3thm4th** [reply]
Thank you, that explanation is perfect. You might want to add that to the Readme for future smooth-brained readers like me.
